homeassistant/components/ssdp/__init__.py

```diff
@@ -559,3 +580,108 @@ def _udn_from_usn(usn: str | None) -> str | None:
+
+    DEVICE_DEFINITION = DeviceInfo(
+        device_type="urn:home-assistant.io:device:HomeAssistant:1",
+        friendly_name="Home Assistant",
+        manufacturer="Home Assistant",
+        manufacturer_url="https://www.home-assistant.io",
+        model_description=None,
+        model_name=APPLICATION_NAME,
+        model_number=current_version,
+        model_url="https://www.home-assistant.io",
+        serial_number=None,
+        udn="filled_later_on",
+        upc=None,
+        presentation_url="https://my.home-assistant.io/",
+        url="/device.xml",
+        icons=[],
+        xml=ET.Element("server_device"),
+    )
+    EMBEDDED_DEVICES: list[type[UpnpServerDevice]] = []
+    SERVICES: list[type[UpnpServerService]] = []
+
+
+class Server:
+    """Class to be visible via SSDP searching and advertisements."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize class."""
+        self.hass = hass
+        self._upnp_servers: list[UpnpServer] = []
+
+    async def async_start(self) -> None:
+        """Start the server."""
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self.async_stop)
+        await self._async_start_ssdp_servers()
+
+    async def _async_get_free_port(self) -> int:
+        """Get a (random) free TCP port."""
+        return 9876
+
+    async def _async_get_instance_udn(self) -> str:
+        """Get Unique Device Name for this instance."""
+        instance_id = await async_get_instance_id(self.hass)
+        return f"uuid:{instance_id[0:8]}-{instance_id[8:12]}-{instance_id[12:16]}-{instance_id[16:20]}-{instance_id[20:32]}".upper()
+
+    async def _async_start_ssdp_servers(self) -> None:
+        """Start the SSDP Listeners."""
+        # Update UDN with our instance UDN.
+        udn = await self._async_get_instance_udn()
+        system_info = await async_get_system_info(self.hass)
+        model_name = system_info["installation_type"]
+        presentation_url = get_url(self.hass)
+        serial_number = await async_get_instance_id(self.hass)
+        HassUpnpServiceDevice.DEVICE_DEFINITION = (
+            HassUpnpServiceDevice.DEVICE_DEFINITION._replace(
+                udn=udn,
+                model_name=model_name,
+                presentation_url=presentation_url,
+                serial_number=serial_number,
+            )
+        )
+
+        # Devices are shared between all sources.
+        for source_ip in await async_build_source_set(self.hass):
+            source_ip_str = str(source_ip)
+            if source_ip.version == 6:
+                source_tuple: AddressTupleVXType = (
+                    source_ip_str,
+                    0,
+                    0,
+                    int(getattr(source_ip, "scope_id")),
+                )
+            else:
+                source_tuple = (source_ip_str, 0)
+            source, target = determine_source_target(source_tuple)
+            http_port = await self._async_get_free_port()
```

Comment:
The feature is meant for people that can’t find their instance. That are not people that configure firewalls I think. However, you’re right, they still want to configure it all perfectly.

Reply:
I don't doubt that. How do you propose this should be configured? Config entry seems a bit of an overkill, so that would leave configuration through `configuration.yaml` as far as I know.

Another options would be to just choose a high port and hard code it. Though there is a chance it will collide with another integration/application/...

Reply:
I'd probably make it a `configuration.yaml` option as it seems like an advanced option.

Also just because we expect people will ask doesn't mean we have to implement it.  I'm only pointing it out in case it makes it easier to design for later in the event you do want to implement it.

Reply:
let's start with just a random port for now 👍

Reply:
It is semi-random now. It tries to get the first free port - on the specific IP - between 40000 and 60000. Feel free to suggest a different range.

Reply:
Lets not overdo it... 40000-40100 it is.